File: factory/Exemplo1/AnimalCachorro.java
public class AnimalCachorro extends Animal {
	public AnimalCachorro(String nome) {
		this.nome = nome;
		System.out.println("Sou o cachorro " + this.nome);
	}    

    public void falar() {
        System.out.println("au au!");
    }    
}

File: factory/Exemplo1/AnimalCachorro.ts
class AnimalCachorro extends Animal {
    constructor(nome: string) {
        super();  // Chama o construtor da classe base
        this.nome = nome;
        console.log("Sou o cachorro " + this.nome);
    }

    falar(): void {
        console.log("au au!");
    }    
}
